### ISSUE TITLE
plot_2d breaks with matplotlib-3.5

### ISSUE THREAD

**Issue by Stefan-Heimersheim** [williamjameshandley/anesthetic#183]
**Describe the bug**
Calling `mcmc.plot_2d(['omegabh2','omegach2','H0'], types={'lower':'kde','diagonal':'kde'});` returns
```
Traceback (most recent call last):
  File "/tmp/test.py", line 15, in <module>
    mcmc.plot_2d(['omegabh2','omegach2','H0'], types={'lower':'kde','diagonal':'kde'});
  File "/home/stefan/rps/codes/anesthetic/anesthetic/samples.py", line 330, in plot_2d
    fig, axes = make_2d_axes(axes, tex=self.tex,
  File "/home/stefan/rps/codes/anesthetic/anesthetic/plot.py", line 296, in make_2d_axes
    axes[x][y].lines = axes[x][y].twin.lines
AttributeError: can't set attribute
```

**To Reproduce**
Install new matplotlib `pip install matplotlib==3.5`, then use `plot_2d`. MWE from plot gallery:
```python
import requests
import tarfile
import matplotlib.pyplot as plt
import numpy as np

for filename in ["plikHM_TTTEEE_lowl_lowE_lensing.tar.gz","plikHM_TTTEEE_lowl_lowE_lensing_NS.tar.gz"]:
    github_url = "https://github.com/williamjameshandley/cosmo_example/raw/master/"
    url = github_url + filename
    open(filename, 'wb').write(requests.get(url).content)
    tarfile.open(filename).extractall()

from anesthetic import MCMCSamples
mcmc_root = 'plikHM_TTTEEE_lowl_lowE_lensing/base_plikHM_TTTEEE_lowl_lowE_lensing'
mcmc = MCMCSamples(root=mcmc_root)
mcmc.plot_2d(['omegabh2','omegach2','H0'], types={'lower':'kde','diagonal':'kde'});
plt.show()
```

**Expected behavior**
Plot works, like with `matplotlib-3.4.1`. Edit: `matplotlib==3.4.3` also works

**Additional context**
python version 3.9.7, anesthetic git version (commit `6f80b6a8b598708ee2409fe461ecfeddce941fb7` from 15th November)

**Comment by williamjameshandley** [comment]
I confirm this is also broken on my machines. The recommended fix for now is to use matplotlib 3.4.3 or lower. 

I'll try and have a look into fixing this once term is over, but it looks like [this breaking API change](https://matplotlib.org/3.5.0/api/prev_api_changes/api_changes_3.5.0.html#axes-children-are-no-longer-separated-by-type) is the cause of the issues.

**Comment by lukashergt** [comment]
Thanks @Stefan-Heimersheim for flagging this up. Should be fixed with #184.
